planner/cascades/transformation_rules.go

```diff
@@ -32,12 +39,135 @@ func GetTransformationRules(node plannercore.LogicalPlan) []Transformation {
 }
 
 var transformationMap = map[memo.Operand][]Transformation{
-	/**
-	operandSelect: []Transformation{
-		nil,
+	memo.OperandSelection: {
+		&PushSelDownTableScan{},
+		&PushSelDownTableGather{},
 	},
-	operandProject: []Transformation{
-		nil,
+	memo.OperandDataSource: {
+		&EnumeratePaths{},
 	},
-	*/
+}
+
+// PushSelDownTableScan pushes the selection down to TableScan.
+type PushSelDownTableScan struct {
+}
+
+// GetPattern implements Transformation interface. The pattern of this rule is `Selection -> TableScan`.
+func (r *PushSelDownTableScan) GetPattern() *memo.Pattern {
+	ts := memo.NewPattern(memo.OperandTableScan)
+	return memo.BuildPattern(memo.OperandSelection, ts)
+}
+
+// Match implements Transformation interface.
+func (r *PushSelDownTableScan) Match(expr *memo.ExprIter) bool {
+	return true
+}
+
+// OnTransform implements Transformation interface.
```

Comment:
how about adding more comments about this method to describe the transformation result, for example:
```go
// OnTransform implements Transformation interface.
//
// It transforms `Sel -> TS` to one of the following new expr:
// 1. `newSel -> newTS`
// 2. `newTS`
//
// Filters of the old `Sel` operator are removed if they are used to calculate
// the key ranges of the `TS` operator.
```